backend/danswer/connectors/cross_connector_utils/html_utils.py

```diff
@@ -57,11 +58,24 @@ def format_document_soup(
                 # Tables are represented in natural language with rows separated by newlines
                 # Can't have newlines then in the table elements
                 element_text = element_text.replace("\n", " ").strip()
+            if last_added_newline and element_text.startswith(" "):
```

Comment:
Some tags are translated to spaces, but in the logic below we make them newlines (example br tag). We don't want to have both newline and space after, it looks ugly!

Reply:
[optional] can we add some examples in the commends here of what this is doing? I think it will help if someone ever has to modify this in the future